vpr/src/base/partition_region.h

```diff
@@ -0,0 +1,38 @@
+#ifndef PARTITION_REGIONS_H
+#define PARTITION_REGIONS_H
+
+#include "region.h"
+#include "atom_netlist_fwd.h"
+
+/**
+ * @file
+ * @brief This file defines the PartitionRegions class. The PartitionRegions class is used to store the union
+ * of regions that a partition can be placed in.
+ */
+
+class PartitionRegion {
+  public:
+    /**
+     * @brief Return the intersection of two PartitionRegions
+     *
+     *   @param part_region     The PartitionRegion that the calling object will be intersected with
+     */
+    PartitionRegion get_intersection(PartitionRegion part_region);
```

Comment:
Lets follow Keith's suggestion and be more like Rect.  
Move this to the bottom of the header file (outside the class), and make it 
PartitionRegion insersection (PartitionRegion a, PartitionRegion b);